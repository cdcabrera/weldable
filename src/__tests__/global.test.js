const global = require('../global');

describe('Global', () => {
  it('should return specific properties', () => {
    expect(global).toMatchSnapshot('specific properties');
  });

  it('should handle errors in multiple formats', () => {
    const errorLikeStr = 'Custom error string';
    const errorObj = new Error('Custom error object');
    const errorLikeObj = { one: errorLikeStr, two: errorObj };
    const errorList = [errorLikeStr, errorObj, errorLikeObj];

    expect(global.errorMessageHandler(errorLikeStr)).toMatchSnapshot('errorLikeStr');
    expect(global.errorMessageHandler(errorObj)).toMatchSnapshot('errorObj');
    expect(global.errorMessageHandler(errorLikeObj)).toMatchSnapshot('errorLikeObj');
    expect(global.errorMessageHandler(errorList)).toMatchSnapshot('errorList');
  });

  it('should determine a promise', () => {
    expect(global.isPromise(Promise.resolve())).toBe(true);
    expect(global.isPromise(async () => {})).toBe(true);
    expect(global.isPromise(() => 'lorem')).toBe(false);
  });

  it('should create a file from a string', () => {
    const { contents, dir, file } = global.createFile('lorem ipsum', {
      dir: './.fixtures'
    });
    expect({ contents, dir, file }).toMatchSnapshot('createFile');
    removeFixture(file);
  });

  it('should set a one-time mutable OPTIONS object', () => {
    const { OPTIONS } = global;
    OPTIONS.lorem = 'et all';
    OPTIONS.dolor = 'magna';
    OPTIONS._set = {
      lorem: 'ipsum',
      sit: function () {
        return `function test ${this.contextPath}`;
      }
    };
    OPTIONS.lorem = 'hello world';
    OPTIONS.dolor = 'sit';

    expect({ isFrozen: Object.isFrozen(OPTIONS), OPTIONS }).toMatchSnapshot('immutable');
  });
});
