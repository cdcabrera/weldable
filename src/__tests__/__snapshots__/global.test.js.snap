// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global should create a file from a string: createFile 1`] = `
{
  "contents": "lorem ipsum",
  "dir": "./.fixtures",
  "file": "80a751fde577028640c419000e33eba6.txt",
}
`;

exports[`Global should handle errors in multiple formats: errorLikeObj 1`] = `
"one: Custom error string
two: Error: Custom error object"
`;

exports[`Global should handle errors in multiple formats: errorLikeStr 1`] = `"Custom error string"`;

exports[`Global should handle errors in multiple formats: errorList 1`] = `
[
  "Custom error string",
  "Custom error object",
  "one: Custom error string
two: Error: Custom error object",
]
`;

exports[`Global should handle errors in multiple formats: errorObj 1`] = `"Custom error object"`;

exports[`Global should return specific properties: specific properties 1`] = `
{
  "OPTIONS": {
    "_set": undefined,
    "contextPath": "../__fixtures__",
  },
  "contextPath": "../__fixtures__",
  "createFile": [Function],
  "dynamicImport": [Function],
  "errorMessageHandler": [Function],
  "isPromise": [Function],
}
`;

exports[`Global should set a one-time mutable OPTIONS object: immutable 1`] = `
{
  "OPTIONS": {
    "contextPath": "../__fixtures__",
    "dolor": "magna",
    "lorem": "ipsum",
    "sit": "function test ../__fixtures__",
  },
  "isFrozen": true,
}
`;
